src/main/java/org/spongepowered/api/world/TeleportHelper.java

```diff
@@ -71,9 +132,16 @@
      * @param location The location to search nearby.
      * @param height The radius of blocks on the y-axis to search.
      * @param width The radius of blocks on the x and z-axis to search.
+     * @param maxDropDistance The number of blocks below a selected block to
+     *                        search for a suitable floor, that is, the maximum
+     *                        distance to a floor that the selected point can
+     *                        be. If this is zero or negative, a floor check
+     *                        will not be performed.
+     * @param params The {@link TeleportHelperKernel} to use to determine if a
+     *               location is safe.
      * @return A safe location near the original location or the original
      *         location if it is deemed safe. If no safe location can be found,
      *         {@link Optional#empty()} will be returned
      */
-    Optional<Location<World>> getSafeLocation(Location<World> location, int height, int width);
+    Optional<Location<World>> getSafeLocation(Location<World> location, int height, int width, int maxDropDistance, TeleportHelperKernel params);
```

Comment:
Simply from a previous iteration when I was cobbling it together, that function signature was something else. Forgot to change it, will get done.